apiserver/facades/client/client/filtering.go

```diff
@@ -282,10 +283,54 @@ func buildUnitMatcherShims(u *state.Unit, patterns []string) []closurePredicate
 
 func matchPortRanges(patterns []string, portRanges ...network.PortRange) (bool, bool, error) {
 	for _, p := range portRanges {
-		pNum, _, _ := strings.Cut(p.String(), "/")
+		var pFrom, pTo int
+		var err error
+		pNumStr, pProto, _ := strings.Cut(p.String(), "/")
+		pFromStr, pToStr, isPortRange := strings.Cut(pNumStr, "-")
+		if isPortRange {
+			pFrom, err = strconv.Atoi(pFromStr)
+			if err != nil {
+				return false, true, err
+			}
+			pTo, err = strconv.Atoi(pToStr)
+			if err != nil {
+				return false, true, err
+			}
+		} else {
+			pFrom, err = strconv.Atoi(pNumStr)
+			if err != nil {
+				return false, true, err
+			}
+			pTo, err = strconv.Atoi(pNumStr)
+			if err != nil {
+				return false, true, err
+			}
+		}
 		for _, patt := range patterns {
-			pattNum, _, _ := strings.Cut(patt, "/")
-			if pNum == pattNum && strings.HasPrefix(p.String(), patt) {
+			var pattFrom, pattTo int
+			pattNumStr, pattProto, isPortPattern := strings.Cut(patt, "/")
+			pattFromStr, pattToStr, isPattRange := strings.Cut(pattNumStr, "-")
+			if isPattRange {
+				pattFrom, err = strconv.Atoi(pattFromStr)
+				if err != nil {
+					return false, true, err
+				}
+				pattTo, err = strconv.Atoi(pattToStr)
+				if err != nil {
+					return false, true, err
+				}
+			} else {
+				pattFrom, err = strconv.Atoi(pattNumStr)
+				if err != nil {
+					return false, true, err
+				}
+				pattTo, err = strconv.Atoi(pattNumStr)
+				if err != nil {
+					return false, true, err
+				}
```

Comment:
Again here. 

Perhaps the duplicated code should be extracted into a function?